### PR TITLE
search_facts: allow fetching full details of found resources

### PR DESCRIPTION
**Pull request by evgeni**
yes, this is missing tests atm :)

### CODE REVIEW

**Comment by sean797** [comment]
:+1: to this -  thanks @evgeni

**Comment by evgeni** [comment]
Added (simple) tests and ready for review :)

File: plugins/modules/foreman_search_facts.py [hunk 1]
```diff
@@ -84,18 +91,26 @@ def main():
...
 
     module.connect()
 
-    response = module.list_resource(entity, search)
+    response = module.list_resource(resource, search)
+
+    if module_params['full_details']:
+        resources = []
+        for res in response:
+            resources.append(module.show_resource(resource, res['id']))
```
Comment (mdellweg) [inline plugins/modules/foreman_search_facts.py]: Why not as a comprehension (modern python)?
`resources = [module.show_resource(resource, res['id'])) for res in response]`
Nice how it it unclear whether res is the abbreviation of resource or response. :smiling_imp:

File: tests/test_playbooks/search_facts.yml [hunk 2]
```diff
@@ -13,6 +13,11 @@
     vars:
       resource: subnets
       return_length: 0
+  - include: tasks/search_facts.yml
+    vars:
+      resource: organizations
+      full_details: true
+      return_length: 2
```
Comment (mdellweg) [inline tests/test_playbooks/search_facts.yml]: Is this `2` a good test result? I think it depends a lot on your test setup. And you did not even include the fixtures for organization, so i guess, 'Test Organization' isn't even one of those.

**Comment by sean797** [comment]
Would you mind also adding something to the EXAMPLE section?

File: plugins/modules/foreman_search_facts.py (hunk 1 above)
Comment (evgeni) [reply plugins/modules/foreman_search_facts.py]: > Nice how it it unclear whether res is the abbreviation of resource or response.

Damn, you noticed it :stuck_out_tongue_closed_eyes:

**Comment by evgeni** [comment]
@mdellweg @sean797 adressed your comments, please have another look :)

**Pull request merged**